server.cpp

```diff
@@ -85,23 +107,55 @@ int main(int argc, char* argv[])
         tcp::resolver resolver(io_service);
         tcp::acceptor acceptor(io_service, *resolver.resolve(tcp::resolver::query(host, port)));
 
+        char buff[1024];
+
+        std::ofstream fout;
+
         for (;;)
         {
             tcp::socket socket(io_service);
             acceptor.accept(socket);
 
-            const std::string message = make_daytime_string();
+            const size_t bytes = read(socket, boost::asio::buffer(buff), std::bind(read_complete, buff, std::placeholders::_1, std::placeholders::_2));
+
+            const std::string msg(buff, bytes);
+
+            const size_t str_end_pos = msg.find("\r");
+            const std::string start_str = msg.substr(0, str_end_pos);
+            const std::string http_version = start_str.substr(str_end_pos - 8);
+            std::string error_message;
+
+            if (start_str.substr(0, 3) != "GET")
+                error_message = http_version + " 405 Method not allowed\r\n";
+            if (http_version != "HTTP/1.1" && http_version != "HTTP/1.0")
+                error_message += http_version + " 505 HTTP Version Not Supported\r\n";
+
+            error_code ignored_error;
+
+            if (!error_message.empty())
+                boost::asio::write(socket, boost::asio::buffer(error_message), ignored_error);
+
+            const std::string date = make_daytime_string();
+            const std::string message = "HTTP/1.1 200 OK\r\nHOST: 127.0.0.1\r\n\r\n" + date;
```

Comment:
I'm not 100%-sure, but I think that the host header should be `Host`, not `HOST`. Please check with the standard.

Also, this header usually contains a hostname (DNS name), not an IP.